arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h

```diff
@@ -21,13 +21,17 @@ static void _mpu_init(void)
 
 	/* Flash region(s): Attribute-0
 	 * SRAM region(s): Attribute-1
+	 * SRAM no cache-able regions(s): Attribute-2
 	 */
 	MPU->MAIR0 =
 		((MPU_MAIR_ATTR_FLASH << MPU_MAIR0_Attr0_Pos) &
 			MPU_MAIR0_Attr0_Msk)
 		|
 		((MPU_MAIR_ATTR_SRAM << MPU_MAIR0_Attr1_Pos) &
-			MPU_MAIR0_Attr1_Msk);
+			MPU_MAIR0_Attr1_Msk)
+		|
+		((MPU_MAIR_ATTR_SRAM_NOCACHE << MPU_MAIR0_Attr2_Pos) &
```

Comment:
We could do that, of course, but, I think it does not matter so much; there is no waste of run-time resources, if MAIR0 has 3 instead of 2 8-bit fields set...

Reply:
If you insist, I can #ifdef it :)

Reply:
Isn't that the same than on ARMv7 when we try to spare the memory regions so that they are available for memory domain and userspace usage?

Reply:
No, in Armv8-m the cacheability attribution is performed through indirection. MAIR0,   MAIR1 contain - together - eight 8-bit fields that store cacheability attributions, and each region can be configured to "point" to any of those 8-bit fields.

So, so far we have 3 fields configured; Flash, SRAM and SRAM-no-cache, so we are good :)